shared/src/commonMain/kotlin/org/hyperskill/app/analytic/domain/processor/AnalyticHyperskillEventProcessor.kt

```diff
@@ -40,6 +42,7 @@ class AnalyticHyperskillEventProcessor(
                 contextMap[PARAM_IS_NOTIFICATIONS_ALLOW] = isNotificationsPermissionGranted
                 contextMap[PARAM_IS_ATT_ALLOW] = isATTPermissionGranted
                 contextMap[PARAM_SCREEN_ORIENTATION] = getScreenOrientationValue(screenOrientation)
+                contextMap[PARAM_IS_INTERNAL_TESTING] = isInternalTesting
```

Comment:
Currently, the method `processEvent` checks if the `PARAM_CONTEXT` key exists in the `resultParams` map or not. So we have 2 branches where we must put values into map, this can lead to bugs like now.

I suggest you to simplify this operation and remove code duplication:
```kotlin
fun processEvent(
    event: AnalyticEvent,
    userId: Long,
    isNotificationsPermissionGranted: Boolean,
    isATTPermissionGranted: Boolean,
    screenOrientation: ScreenOrientation,
    isInternalTesting: Boolean
): HyperskillProcessedAnalyticEvent {
    val resultParams = event.params.toMutableMap()

    val contextMap = when (resultParams.containsKey(PARAM_CONTEXT)) {
        true -> resultParams[PARAM_CONTEXT].safeCast<Map<String, Any>>()?.toMutableMap()
        false -> mutableMapOf()
    } ?: mutableMapOf()

    updateContextMap(
        contextMap,
        isNotificationsPermissionGranted = isNotificationsPermissionGranted,
        isATTPermissionGranted = isATTPermissionGranted,
        screenOrientation = screenOrientation,
        isInternalTesting = isInternalTesting
    )
    resultParams[PARAM_CONTEXT] = contextMap
    resultParams[PARAM_USER] = userId

    return HyperskillProcessedAnalyticEvent(name = event.name, params = resultParams.toMap())
}

private fun updateContextMap(
    contextMap: MutableMap<String, Any>,
    isNotificationsPermissionGranted: Boolean,
    isATTPermissionGranted: Boolean,
    screenOrientation: ScreenOrientation,
    isInternalTesting: Boolean
) {
    contextMap.apply {
        this[PARAM_PLATFORM] = platform.analyticName
        this[PARAM_IS_NOTIFICATIONS_ALLOW] = isNotificationsPermissionGranted
        this[PARAM_IS_ATT_ALLOW] = isATTPermissionGranted
        this[PARAM_SCREEN_ORIENTATION] = getScreenOrientationValue(screenOrientation)
        this[PARAM_IS_INTERNAL_TESTING] = isInternalTesting
    }
}
```